bindings/python/tests/basic/test-py-attribute.py

```diff
@@ -0,0 +1,106 @@
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+import OpenGeode_py_basic as basic
+
+def test_constant_attribute( manager ):
+    constant_attribute = manager.find_or_create_attribute_constant_bool( "bool", True )
+    
+    attribute = manager.find_attribute_bool( "bool" )
+    if not attribute.value( 0 ):
+        raise ValueError( "[Test] Should be equal to True" )
+    
+    constant_attribute.set_value( False )
+    if attribute.value( 12 ):
+        raise ValueError( "[Test] Should be equal to False" )
+
+def test_int_variable_attribute( manager ):
+    variable_attribute = manager.find_or_create_attribute_variable_int( "int", 12 )
+    variable_attribute.set_value( 3, 3 )
+
+    attribute = manager.find_attribute_int( "int" )
+    if attribute.value( 3 ) != 3:
+        raise ValueError( "[Test] Should be equal to 3" )
+    if attribute.value( 6 ) != 12:
+        raise ValueError( "[Test] Should be equal to 12" )
+
+    variable_attribute.set_value( 3, 5 )
+    if attribute.value( 3 ) != 5:
+        raise ValueError( "[Test] Should be equal to 5" )
+
+def test_double_sparse_attribute( manager ):
+    sparse_attribute = manager.find_or_create_attribute_sparse_double( "double", 12 )
+    sparse_attribute.set_value( 3, 3 )
+    sparse_attribute.set_value( 7, 7 )
+
+    attribute = manager.find_attribute_double( "double" )
+    if attribute.value( 3 ) != 3:
+        raise ValueError( "[Test] Should be equal to 3" )
+    if attribute.value( 6 ) != 12:
+        raise ValueError( "[Test] Should be equal to 12" )
+    if attribute.value( 7 ) != 7:
+        raise ValueError( "[Test] Should be equal to 7" )
+
+    sparse_attribute.set_value( 3, 5 )
+    if attribute.value( 3 ) != 5:
+        raise ValueError( "[Test] Should be equal to 5" )
+
+def test_number_of_attributes( manager, nb ):
+    if len(manager.attribute_names()) != nb:
+        raise ValueError( "[Test] Not the correct number of attributes in the manager" )
+
+def test_delete_attribute_elements( manager ):
+    to_delete = [False] * manager.nb_elements()
+    to_delete[3] = True
+    to_delete[5] = True
+    manager.delete_elements( to_delete )
+    if manager.nb_elements() != len(to_delete) - 2:
+        raise ValueError( "[Test] Two attribute elements should have being removed" )
+
+def test_sparse_attribute_after_element_deletion( manager ):
+    sparse_attribute = manager.find_attribute_double( "double" )
+    if sparse_attribute.value( 0 ) != 12:
+        raise ValueError( "Element 0 of sparse attribute should be 12 " )
+    if sparse_attribute.value( 5 ) != 7:
+        raise ValueError( "Element 5 of sparse attribute should be 7 " )
+    if sparse_attribute.value( 7 ) != 12:
+        raise ValueError( "Element 7 of sparse attribute should be 12 " )
+
+manager = basic.AttributeManager()
```

Comment:
if __name__ == "__main__" ? To add in all the tests

Reply:
When this "if" could be false ? It is a test not a file to import. But I can add it by safety.

Reply:
It means you can import one test in another to avoid declaration duplication. Maybe very rare but may be usefull. What do you think ?